arestor/api/v1/openstack.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2016 Cloudbase Solutions Srl
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""Arestor API endpoint for OpenStack Mocked Metadata."""
+import json
+import base64
+import cherrypy
+
+from oslo_log import log as logging
+
+from arestor.api import base as base_api
+from arestor.common import exception
+
+
+LOG = logging.getLogger(__name__)
+
+
+class _OpenStackResource(base_api.Resource):
+    """Base class for OpenStack resources."""
+
+    exposed = True
+
+    def _get_openstack_data(self, name, field=None):
+        """Retrieve the required resource from the Openstack namespace."""
+        data = None
```

Comment:
This is a good question. Where is no date do we give an empty string or say it's `None`.